pkg/report/github_sbom.go

```diff
@@ -0,0 +1,185 @@
+}
+
+type GithubSbomWriter struct {
+	Output  io.Writer
+	Version string
+}
+
+func init() {
+	CustomTemplateFuncMap["now"] = time.Now
+	CustomTemplateFuncMap["getenv"] = os.Getenv
+}
+
+func (gsbmw GithubSbomWriter) Write(report types.Report) error {
+	getenv, ok := CustomTemplateFuncMap["getenv"].(func(string) string)
+	if !ok {
+		return xerrors.Errorf("invalid getenv reference")
+	}
+	githubSbom := &GithubSbom{}
+
+	//use now() method that can be overwritten while integration tests run
+	githubSbom.Scanned = CustomTemplateFuncMap["now"].(func() time.Time)().Format(time.RFC3339)
+	githubSbom.Detector = GithubSbomDetector{
+		Name:    "trivy",
+		Version: gsbmw.Version,
+		Url:     "https://github.com/aquasecurity/trivy",
+	}
+	githubSbom.Version = 1 // The version of the repository snapshot submission. It's not clear what value to set
+
+	githubSbom.Ref = getenv("GITHUB_REF")
+	githubSbom.Sha = getenv("GITHUB_SHA")
+
+	githubSbom.Job = &GithubSbomJob{
+		Correlator: (fmt.Sprintf("%s_%s", getenv("GITHUB_WORKFLOW"), getenv("GITHUB_JOB"))),
+		Id:         getenv("GITHUB_RUN_ID"),
+	}
+
+	githubSbom.Metadata = getMetadata(report)
+
+	manifests := make(map[string]GithubSbomManifest)
+
+	for _, result := range report.Results {
+		manifest := GithubSbomManifest{}
+		manifest.Name = result.Type
+		//show path for languages only
+		if result.Class == types.ClassLangPkg {
+			manifest.File = &GithubSbomFile{
+				SrcLocation: result.Target,
+			}
+		}
+		if result.Packages == nil {
+			return xerrors.Errorf("unable to find packages")
+		}
+
+		resolved := make(map[string]GithubSbomPackage)
+
+		for _, pkg := range result.Packages {
+			var err error
+			githubSbomPkg := GithubSbomPackage{}
+			githubSbomPkg.Scope = RuntimeScope
+			githubSbomPkg.Relationship = getPkgRelationshipType(pkg)
+			githubSbomPkg.Dependencies = getDependencies(report.Results, pkg)
+			githubSbomPkg.PackageUrl, err = buildPurl(result.Type, pkg)
+			if err != nil {
+				return xerrors.Errorf("unable to build purl: %w for the package: %s", err, pkg.Name)
+			}
+
+			resolved[pkg.Name] = githubSbomPkg
+		}
+
+		manifest.Resolved = resolved
+		manifests[result.Target] = manifest
+	}
+
+	githubSbom.Manifests = manifests
+
+	output, err := json.MarshalIndent(githubSbom, "", "  ")
+	if err != nil {
+		return xerrors.Errorf("failed to marshal github sbom: %w", err)
+	}
+
+	if _, err = fmt.Fprint(gsbmw.Output, string(output)); err != nil {
+		return xerrors.Errorf("failed to write github sbom: %w", err)
+	}
+	return nil
+}
+
+func getMetadata(report types.Report) Metadata {
+	metadata := Metadata{}
+	if report.Metadata.RepoTags != nil {
+		metadata["aquasecurity:trivy:RepoTag"] = report.Metadata.RepoTags[0]
+	}
+	if report.Metadata.RepoDigests != nil {
+		metadata["aquasecurity:trivy:RepoDigest"] = report.Metadata.RepoDigests[0]
+	}
+	return metadata
+}
+
+func getDependencies(results []types.Result, pkg ftypes.Package) []string {
```

Comment:
Only direct dependencies are in `DependsOn`. I'm not sure if it's enough.
Also in another PR I'm adding dependencies a bit different way
https://github.com/aquasecurity/trivy/pull/1970/files#diff-69cf3117d3866088e0326f2f36c576ede844b8f0b6bd9a2d966b8a7dc5f4f41aR238

Anyway most of dependencies are empty at the moment because of missed implementation in `go-dep-parser`

Reply:
- package a
  - package b
    - package c

In this case, `DependsOn` of `package a` contains only `package b`, not contains `package c` since it is transitive dependency. But `package b` contains `package c` in `DependsOn`. I think it is fine, but could you confirm it? @ShiraCohen33

Reply:
Yes, do you want that I'll ensure it in GitHub Slack?

Reply:
yes, please